CONTRIBUTING.md

```diff
@@ -57,6 +57,81 @@ You can build and start a local docs website by running these two commands.
   - `npm run docs-buildDockerImage` OR `docker build -t squidfunk/mkdocs-material ./docs/`
   - `npm run docs-runLocalDocker` OR `docker run --rm -it -p 8000:8000 -v ${PWD}:/docs squidfunk/mkdocs-material`
 
+### Tests
+
+Tests are under `tests` folder of each modules and split into two categories: unit tests and e2e tests.
+
+You can run each group separately or all together thanks to [jest-runner-groups](https://www.npmjs.com/package/jest-runner-groups).
+
+Unit tests, under `tests/unit` folder are standard [Jest](https://jestjs.io) tests.
+
+End-to-end tests, under `tests/e2e` folder, will test the module features by deploying Lambdas functions into your AWS Account. We use CDK lib for Typescript for creating infrastructure, and `aws sdk` for invoking the functions and assert on the expected behaviors. All steps are also executed by Jest.
+
+Running end-to-end tests will deploy AWS resources. You will need an AWS account and the tests might incur costs. The cost from **some services** are covered by the [AWS Free Tier](https://aws.amazon.com/free/?all-free-tier.sort-by=item.additionalFields.SortRank&all-free-tier.sort-order=asc&awsf.Free%20Tier%20Types=*all&awsf.Free%20Tier%20Categories=*all) but not all of them. If you don't have an AWS Account follow [these instructions to create one](https://aws.amazon.com/premiumsupport/knowledge-center/create-and-activate-aws-account/).
+
+When contributing to this repository, these end-to-end tests are run by the maintainers before merging a PR.
+
+
+**Unit testing**
+
+**Write**
+
+As mentioned before, tests are split into groups thanks to [jest-runner-groups](https://www.npmjs.com/package/jest-runner-groups) and therefore each test needs to be tagged properly by adding the following comments in the header of your unit test file:
+
+```typescript
+/**
+ * Tests metrics
+ *
+ * @group unit/<YOUR CATEGORY>/<YOUR SUB CATEGORY>
+ */
+```
+
+**Run**
+
+ `npm run test`
```

Comment:
For consistency with the e2e test, can we change this command to be everywhere:
```
npm run test:unit
```

So that in the future we can:
- create 1 line command that runs all the tests in sequence:
`npm run test` will execute the unit tests and then the e2e tests. If we have new kind of tests we want to add to the sequence, we can do it by adding them here.
I used this similar pattern in past teams as a customer and makes things easy.
- add new types of tests with related name if we need (`npm run test:acceptance`)